Proxy update (MU) (#740)
SUSE/doc-susemanager: modules/upgrade/pages/proxy-upgrade.adoc

[[update.suse.manager.proxy]]
= Update {productname} Proxy

> {productname} Proxies are managed as clients by {productname}.

Reviewer: ```suggestion
{productname} Proxies are managed in the same way as clients.
```

> Thus patches and package updates combined as maintenance updates (MU) are to be installed with {productname} as it is done with other {productname} clients.

Reviewer: ```suggestion
Maintenance updates (MU) can be installed on a {productname} Proxy in the same way as other clients.
```

> When the update is installed, you must check whether a restart of the proxy service is required.

Reviewer: ```suggestion
Some updates will require a restart of the proxy service.
```

// a remote command (or a salt state) can perform the restart.
// Creating an Action Chain can help with this task.
// With an action chain you can update proxies on batches, or all at the same time:
// These are the basic steps to run:
////
1. Run a salt command: `spacewalk-proxy stop`
2. Update the packages
3. Run a salt command: `spacewalk-proxy start`
////

> == Before You Begin

Reviewer: ```suggestion
== Prepare to Upgrade
```

We really need to standardise this heading across the suite. We use lots of different wording for pre-reqs.

Before you perform any proxy upgrade, schedule a maintenance window.

> The clients registered to {productname} via the proxy will not be able checking in with {productname} for that period of time.

Reviewer: ```suggestion
The clients registered to {productname} through the proxy will not be able to connect to {productname} while the upgrade is in progress.
```

For more information about maintenance windows, see xref:administration/maintenance-window.adoc[].

== Update the Proxy

> 

Author: ```suggestion

To update a proxy you first stop the proxy service, then update the software and finally restart the proxy service.

```

> 

Reviewer: Can we add a line in here?

Author: > Can we add a line in here?

A sentence is in.

.Procedure: Updating the {productname} Proxy

> . From the {productname} Server, where the Proxy is registered as a client, send the [command]``spacewalk-proxy stop`` command.

Reviewer: ```suggestion
. On the {productname} Server where the proxy is registered as a client, stop the proxy service:
+
----
spacewalk-proxy stop
----
```

> . Select the packages to be updated on the Proxy.

Reviewer: ```suggestion
. Select the packages to be updated on the proxy.
```

> . From the {productname} Server, where the Proxy is registered as a client, send the [command]``spacewalk-proxy start`` command.

Reviewer: ```suggestion
. On the {productname} Server where the proxy is registered as a client, start the proxy service:
+
----
spacewalk-proxy start
----
```

> It is recommended to create an action chain of this command sequence.

Reviewer: ```suggestion
If you need to update many proxies, you can create an action chain of this command sequence.
```

> Then you can apply it to several proxies to update them in batches or all at the same time.

Reviewer: ```suggestion
You can use the action chain to perform upgrades on multiple proxies at the same time.
```

[[update.suse.manager.proxy]]

> = Update {productname} Proxy

Reviewer 2: What's the difference between an 'update', an 'upgrade' and a 'migration'?

I think we need consistent naming. Now the file is `proxy-upgrade`, but this title is `Update`, and then bellow we talk about `updates`.

Checking the server (https://github.com/SUSE/doc-susemanager/blob/f07fff0d73a909dd29bb42b2cfe60a4a082f824f/modules/upgrade/pages/server-upgrade.adoc), it seems we have the same problem.

So FMPOV: let's choose one convention (updating and migrating, or upgrading and migrating), and then let's use it everywhere.

Author: I intentionally used the "wrong" filename here as well.  Yes, later we should change both filenames at once.

Reviewer 2: ACK.

{productname} Proxies are managed in the same way as clients.
Maintenance updates (MU) can be installed on a {productname} Proxy in the same way as other clients.

> Some updates will require a restart of the proxy service.

Reviewer 2: If we are talking in the context of a MU for the SUSE Manager Proxy product... it will **always** require a restart. Not sure how can express this.

Reviewer: ```suggestion
MU updates require a restart of the proxy service.
```

// a remote command (or a salt state) can perform the restart.
// Creating an Action Chain can help with this task.
// With an action chain you can update proxies on batches, or all at the same time:
// These are the basic steps to run:
////
1. Run a salt command: `spacewalk-proxy stop`
2. Update the packages
3. Run a salt command: `spacewalk-proxy start`
////

> == Prepare to Upgrade

Reviewer 2: Here is "upgrade"

> == Prepare to Upgrade

Author: ```suggestion
== Prepare to Update
```

> Before you perform any proxy upgrade, schedule a maintenance window.

Reviewer 2: Should be "proxy update"

Author: yes

> The clients registered to {productname} through the proxy will not be able to connect to {productname} while the upgrade is in progress.

Reviewer 2: **while the upgrade** -> **while the update**

For more information about maintenance windows, see xref:administration/maintenance-window.adoc[].

> == Update the Proxy

Reviewer 2: Here is "update".

Author: should be update here, IMO.

To update a proxy you first stop the proxy service, then update the software and finally restart the proxy service.

.Procedure: Updating the {productname} Proxy

. On the {productname} Server where the proxy is registered as a client, stop the proxy service:
+
----
spacewalk-proxy stop
----

> . Select the packages to be updated on the proxy.

Reviewer 2: Shouln't we just say "Update the packages on the proxy" to be more generic?

Otherwise the user should "select the packages", and then "apply the selection".

Reviewer: I vote for the latter. In a procedure, it's all about the verbs :wink:

Author: done

. On the {productname} Server where the proxy is registered as a client, start the proxy service:
+
----
spacewalk-proxy start
----

If you need to update many proxies, you can create an action chain of this command sequence.

> You can use the action chain to perform upgrades on multiple proxies at the same time.

Reviewer 2: Not "upgrades", but "updates".

Author: oops, yes
